File: src/test/kotlin/domain/TextUtilsPreProcessTest.java
package domain;

import org.assertj.core.api.Assertions;
import org.junit.Assert;
import org.junit.Test;

import java.util.Arrays;
import java.util.List;

/**
 * Created by Zahar on 16.05.16.
 */
public class TextUtilsPreProcessTest {

	@Test
	public void testComplex() {
		List<String> words = Arrays.asList("qwer", "asdf","|qwer","or","qwer|", "zxc");

		List<String> wordsqwer = TextUtils.Companion.postProcessText(words);
		Assertions.assertThat(wordsqwer)
			.hasSize(4)
			.contains("qwer", "asdf","qwer or qwer","zxc");
	}
}

File: src/test/kotlin/domain/TextUtilsPreProcessTest.kt
package domain

import org.junit.Assert
import org.junit.Test

import java.util.Arrays

/**
 * Created by Zahar on 16.05.16.
 */
class TextUtilsPreProcessTest {

	@Test
	fun testComplex() {
		val rawText = "qwer asdf qwer or qwer zxc"
		val dictionary = Arrays.asList("qwer or qwer", "qwer asdf")

		val strings = TextUtils.preProcessText(dictionary, rawText)
		Assert.assertEquals("|qwer asdf| |qwer or qwer| zxc", strings)
	}
}
